### PR TITLE
Pixiecore

### PR DESCRIPTION
**Pull request by xnaveira**
These are changes to be able to use pixiecore in api mode instead for foreman proxy.

No need to merge, just FYI

X

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -66,22 +65,14 @@ func buildHandler(response http.ResponseWriter, request *http.Request, config Co
...
-	err = m.setBuildMode(config, template)
+	err = m.setBuildMode(config)
 	if err != nil {
 		log.Println(err)
 		http.Error(response, fmt.Sprintf("Failed to set build mode on %s", hostname), 500)
 		return
 	}
 
-	config.MachineState[hostname] = "Installing"
+	log.Println(m)
```
Comment (jhaals) [inline main.go]: This is not needed.

File: main.go [hunk 2]
```diff
@@ -138,6 +129,33 @@ func status(response http.ResponseWriter, request *http.Request, config Config)
 	response.Write(result)
 }
 
+func pixieHandler(response http.ResponseWriter, request *http.Request, config Config) {
+
+	macaddr := mux.Vars(request)["macaddr"]
+	hostname, found := config.MachineBuild[macaddr]
```
Comment (jhaals) [inline main.go]: should be hostname, err

File: main.go [hunk 3]
```diff
@@ -138,6 +129,33 @@ func status(response http.ResponseWriter, request *http.Request, config Config)
...
+
+	if found == false {
```
Comment (jhaals) [inline main.go]: if err != nil

Comment (xnaveira) [inline main.go]: Yeah, I tried but a map lookup gives you back true or false, not an error, hence the found.

Comment (jhaals) [inline main.go]: You're correct!

**Comment by jhaals** [comment]
:star:

**Pull request merged**